src/com/tfl/billing/JourneyManager.java

```diff
@@ -0,0 +1,45 @@
+package com.tfl.billing;
+
+import com.tfl.external.Customer;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.UUID;
+
+public class JourneyManager implements TripManager {
+
+    private final List<JourneyEvent> eventLog;
+
+    public JourneyManager(List<JourneyEvent> events) {
+        this.eventLog = events;
+    }
+
+    List<JourneyEvent> getJourneyEvents(Customer customer) {
+        List<JourneyEvent> customerJourneyEvents = new ArrayList<>();
+        for (JourneyEvent journeyEvent : eventLog) {
+            if (journeyEvent.cardId().equals(customer.cardId())) {
+                customerJourneyEvents.add(journeyEvent);
+            }
+        }
+        return customerJourneyEvents;
+    }
+
+    public List<Journey> getJourneys(Customer customer) {
```

Comment:
I would like to implement penalty charges. The easiest way to do it would be to throw an exception if this method encounters a JourneyStart without an end or a JourneyEnd without a start. Will this destroy any logic down the road?

Reply:
https://github.com/seriousbee/Oyster/blob/07c171792d7b01ae1efa87ffb912c1e917656cee/src/com/tfl/billing/TravelTracker.java#L139-L157